docs/components/table-view.md

```diff
@@ -0,0 +1,1016 @@
+  columnVisibility?: Record<string, boolean> // column visibility, if visibility is toggleable
+}
+```
+
+<KTableView
+  :table-preferences="{ columnVisibility: { email: false } }"
+  :data="basicData"
+  :headers="basicHeaders(false, null, 'email')"
+/>
+
+```html
+<KTableView
+  :table-preferences="{ columnVisibility: { email: false } }"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### rowAttrs
+
+Function for adding custom attributes to each row. The function should return an object of type `TableRowAttributes`:
+
+```ts
+interface TableRowAttributes {
+  [key: string]: string | RouteLocationRaw | undefined
+  /** RouteLocationRaw or url string for row link */
+  to?: RouteLocationRaw | string
+  /** Target for row link */
+  target?: '_self' | '_blank' | '_parent' | '_top'
+}
+```
+
+The passed function receives row value object as an argument.
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders()"
+  :row-attrs="(row) => { return { 'data-testid': `row-${row.id}` } }"
+/>
+
+```html
+<KTableView
+  :row-attrs="(row) => { return { 'data-testid': `row-${row.id}` } }"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+#### Row links
+
+You can pass optional `to: RouteLocationRaw | string` parameter in the returned object to turn the entire row into a link.
+
+<KTableView
+  :row-attrs="getRowLinks"
+  :data="basicData"
+  :headers="basicHeaders()"
+/>
+
+```vue
+<template>
+  <KTableView
+    :row-attrs="getRowAttrs"
+    :data="tableData"
+    :headers="headers"
+  />
+</template>
+
+
+<script setup lang="ts">
+import type { TableRowAttributes } from '@kong/kongponents'
+
+const headers = [...]
+
+const tableData = [...]
+
+const getRowAttrs = (row: Record<string, any>): TableRowAttributes => ({
+  // using static route for demonstration purposes
+  // but you can generate dynamic routes based on the row data
+  to: { name: 'home' }, // pass a string to render the link as an anchor element instead of a router-link
+  target: '_blank',
+})
+</script>
+```
+
+### cellAttrs
+
+Function for adding custom attributes to each table cell. The function should return an object with `key: value` pairs for each attribute.
+
+The passed function receives an object with these parameters as an argument:
+```ts
+{
+  headerKey: string // header key
+  row: object // row value
+  rowIndex: number // row index
+  colIndex: index // column index
+}
+```
+
+<KTableView
+  :data="basicData"
+  :headers="basicHeaders()"
+  :cell-attrs="({ headerKey, row }) => { return { 'data-testid': `column-${headerKey}-row-${row.id}` } }"
+/>
+
+```html
+<KTableView
+  :cell-attrs="({ headerKey, row }) => { return { 'data-testid': `column-${headerKey}-row-${row.id}` } }"
+  :data="tableData"
+  :headers="headers"
+/>
+```
+
+### maxHeight
+
+Pass a number (px) you want to limit table height to. If the height table exceeds the specified number, it will be scrollable. Table header is a `position: sticky;` element and will always be visible.
```

Comment:
question: is the table header **always** sticky?

Reply:
Yes. Doesn't matter whether the height is limited or now - that makes the logic simpler

Reply:
I'm wondering if we may end up with some issues with the sticky header in Konnect, but I can't remember what area I was thinking this could be an issue

Reply:
It's been sticky in KTable since... at least reskin but maybe forever

Reply:
ah 😅